backend/core/src/main/java/org/sonarsource/sonarlint/core/ServerApiProvider.java

```diff
@@ -137,4 +145,44 @@ private HttpClient getClientFor(EndpointParams params, Either<TokenDto, Username
       userPass -> httpClientProvider.getHttpClientWithPreemptiveAuth(userPass.getUsername(), userPass.getPassword()));
   }
 
+  @Override
+  public ServerConnection getConnectionOrThrow(String connectionId) {
+    var serverApi = getServerApiOrThrow(connectionId);
+    return new ServerConnection(connectionId, serverApi, client);
+  }
+
+  public Optional<ServerConnection> tryGetConnection(String connectionId) {
+    return getServerApi(connectionId)
+      .map(serverApi -> new ServerConnection(connectionId, serverApi, client));
+  }
+
+  public Optional<ServerConnection> tryGetConnectionWithoutCredentials(String connectionId) {
+    return getServerApiWithoutCredentials(connectionId)
+      .map(serverApi -> new ServerConnection(connectionId, serverApi, client));
+  }
+
+  @Override
+  public ServerApi getTransientConnection(String token,@Nullable String organization,  String baseUrl) {
+    return getServerApi(baseUrl, organization, token);
+  }
+
+  @Override
+  public void withValidConnection(String connectionId, Consumer<ServerApi> serverApiConsumer) {
+    getValidConnection(connectionId).ifPresent(connection -> connection.withClientApi(serverApiConsumer));
+  }
+
+  @Override
+  public <T> Optional<T> withValidConnectionAndReturn(String connectionId, Function<ServerApi, T> serverApiConsumer) {
+    return getValidConnection(connectionId).map(connection -> connection.withClientApiAndReturn(serverApiConsumer));
+  }
+
+  @Override
+  public <T> Optional<T> withValidConnectionFlatMapOptionalAndReturn(String connectionId, Function<ServerApi, Optional<T>> serverApiConsumer) {
+    return getValidConnection(connectionId).map(connection -> connection.withClientApiAndReturn(serverApiConsumer)).flatMap(Function.identity());
+  }
+
+  @VisibleForTesting
+  public Optional<ServerConnection> getValidConnection(String connectionId) {
+    return tryGetConnection(connectionId).filter(ServerConnection::isValid);
+  }
```

Comment:
This is used only in a single test, couldn't we get rid of it and use another method?